lnwallet/channel.go

```diff
@@ -1736,20 +1743,15 @@ func (lc *LightningChannel) CancelHTLC(rHash [32]byte) (uint32, error) {
 	defer lc.Unlock()
 
 	var addEntry *PaymentDescriptor
-	for e := lc.theirUpdateLog.Front(); e != nil; e = e.Next() {
-		htlc := e.Value.(*PaymentDescriptor)
-		if htlc.EntryType != Add {
-			continue
-		}
 
-		if !htlc.settled && bytes.Equal(htlc.RHash[:], rHash[:]) {
-			addEntry = htlc
-			break
-		}
+	if _, ok := lc.rHashMap[rHash]; ok {
+		addEntry = lc.rHashMap[rHash][0]
 	}
+
```

Comment:
With the change implemented in this PR, this new fragment and the one above in `SettleHTLC` can be re-written as follows (psuedo-go-code): 
```go
<remove the addEntry and targetHTLC variables> 
[addEntry|targetHTLC], ok := lc.rHashMap[<var from earlier>]
if !ok {
    return an error now
}
```